### PR TITLE
Add -kube suffix to pillar when building for kubevirt-EVE

### PR DESCRIPTION
**Pull request by milan-zededa**
When pillar is built for kubevirt-EVE (`HV=kubevirt`), it carries additional kubernetes-specific code and dependencies. It is important to distinguish it from pillar built for kvm/xen-EVE. Otherwise, previously built kvm/xen pillar can be accidentally taken from the linuxkit cache and used for a kubevirt-EVE image, missing all those kubernetes dependencies.

### CODE REVIEW

File: tools/parse-pkgs.sh [hunk 1]
```diff
@@ -17,21 +17,36 @@ get_git_tag() {
...
+    local pkg="$1"
+    _linuxkit_tag 1 0 "${pkg}"
+}
+
+linuxkit_kubevirt_dev_tag() {
+    local pkg="$1"
+    _linuxkit_tag 1 1 "${pkg}"
 }
 
 _linuxkit_tag() {
```
Comment (eriknordmark) [inline tools/parse-pkgs.sh:38]: Would it make sense to add a comment to document the three arguments and their types?

**Review by eriknordmark — COMMENTED** [review]
Today we only have conditional compilation for kubevirt in pillar, but this might grow over time. Is there some (*BUILD*.md) where it makes sense documenting this at a high level? (And I don't know if we have ARM conditional compiles for other pkgs - if so maybe @rene should make sure that is document as well.)

**Comment by milan-zededa** [comment]
> Today we only have conditional compilation for kubevirt in pillar, but this might grow over time. Is there some (_BUILD_.md) where it makes sense documenting this at a high level? (And I don't know if we have ARM conditional compiles for other pkgs - if so maybe @rene should make sure that is document as well.)

Added some documentation to BUILD.md

**Review by eriknordmark — APPROVED** [review]
LGTM. One nit about the doc.

File: docs/BUILD.md [hunk 2]
```diff
@@ -502,6 +502,22 @@ The package `pillar` contains, unsurprisingly, the `pillar` services that are re
...
+
+* `make pkg/pillar`: Production version of pillar for the KVM or Xen hypervisor. There is no difference
+  in the pillar build between KVM and Xen hypervisors, and the default `HV=kvm` can be used for both.
+* `make pkg/pillar DEV=y`: Development version of pillar for the KVM or Xen hypervisor. Debug symbols
+  are preserved and seccomp is disabled in the pillar container.
+* `make pkg/pillar HV=kubevirt`: Production version of pillar for the KubeVirt hypervisor (EVE using
+  K3s + KubeVirt to deploy applications as VMs inside Kubernetes Pods). Contains additional
+  microservices and Go package dependencies specific to Kubernetes.
+* `make pkg/pillar HV=kubevirt DEV=y`: Development version of pillar for the KubeVirt hypervisor.
+  Debug symbols are preserved and seccomp is disabled in the pillar container.
```
[the remark this answers is not in the document]
Comment (milan-zededa) [reply docs/BUILD.md:519]: The idea was probably just to be able to debug zedbox. We didn't care about other microservices at the time when DEV was introduced.
Btw. seccomp is disabled here:
https://github.com/lf-edge/eve/blob/master/pkg/pillar/build-dev.yml#L27-L28
https://github.com/lf-edge/eve/blob/master/pkg/pillar/build-kubevirt-dev.yml#L27-L28

File: docs/BUILD.md [hunk 3]
```diff
@@ -502,6 +502,22 @@ The package `pillar` contains, unsurprisingly, the `pillar` services that are re
 
 `pillar` depends upon the latest versions of these being available at its compile time in its vendor directory at [pkg/pillar/vendor](../pkg/pillar/vendor). The target `make proto-vendor` will vendor them into [pkg/pillar/vendor](../pkg/pillar/vendor).
 
+##### pillar build variants
+
+The package pillar is built using the Makefile command: `make pkg/pillar [VAR1=value1 VAR2=value2 ...]`
+
+Depending on the values of optional variables, the following pillar build variants are currently available:
+
+* `make pkg/pillar`: Production version of pillar for the KVM or Xen hypervisor. There is no difference
```
[the remark this answers is not in the document]
Comment (milan-zededa) [reply docs/BUILD.md:511]: True but this is for the pillar container where the output name+tag are the same.
Btw. I have added output name for the each pillar container variant into the documentation.

File: docs/BUILD.md [hunk 4]
```diff
@@ -502,6 +502,29 @@ The package `pillar` contains, unsurprisingly, the `pillar` services that are re
...
+  K3s + KubeVirt to deploy applications as VMs inside Kubernetes Pods). Contains additional
+  microservices and Go package dependencies specific to Kubernetes.
+  The name of the built pillar container is `lfedge/eve-pillar-kube:<tag>`.
+* `make pkg/pillar HV=kubevirt DEV=y`: Development version of pillar for the KubeVirt hypervisor.
+  Debug symbols are preserved and seccomp is disabled in the pillar container.
+  The name of the built pillar container is `lfedge/eve-pillar-kube-dev:<tag>`.
+
+Only `lfedge/eve-pillar` is currently published to Dockerhub as part of `lf-edge/eve` Github actions
+for the master branch and EVE releases.
+
```
Comment (milan-zededa) [inline docs/BUILD.md:527]: I have also added this note about publishing.

File: docs/BUILD.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (eriknordmark) [reply docs/BUILD.md:511]: Ah - sorry for my confusion,

**Pull request merged**